server/etcdmain/config.go

```diff
@@ -62,6 +62,17 @@ var (
 		"test.coverprofile",
 		"test.outputdir",
 	}
+
+	deprecatedFlags = map[string]string{
+		"enable-v2":                "--enable-v2 is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"experimental-enable-v2v3": "--experimental-enable-v2v3 is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"proxy":                    "--proxy is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"proxy-failure-wait":       "--proxy-failure-wait is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"proxy-refresh-interval":   "--proxy-refresh-interval is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"proxy-dial-timeout":       "--proxy-dial-timeout is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"proxy-write-timeout":      "--proxy-write-timeout is deprecated in 3.5 and will be decommissioned in 3.6.",
+		"proxy-read-timeout":       "--proxy-read-timeout is deprecated in 3.5 and will be decommissioned in 3.6.",
```

Comment:
maybe we can consider using pattern to generate message, like `%s is deprecated in 3.5 and will be decommissioned in 3.6`. Right now, it looks good.

Reply:
Intentionally to keep the flexibility at the cost of duplication. In a followup PR, we will add the flag --v2-deprecation and set a different message.

Reply:
We will only process `--v2-deprecation` in 3.6 for now, so updated per the suggestion.